### DISCUSSION TITLE
Wait useQuery in server

### DISCUSSION THREAD

**Discussion by PetrVasilev** [vercel/next.js#16321] [Help]
I did exactly like in with-apollo example. But useQuery for getting my user info is working only in client side. How i can wait this query?

**Comment by charliewilco** [comment]
instead of using the hook inside of your component on the server, what you might want to try is creating an instance of the client inside of `getServerSideProps` or `getStaticProps` and use it like this:

```js

const getStaticProps = async () => {
   const client = createApolloClient();
   client.query(/* Same parameters as useQuery */);
  return {
    props: {
      initialApolloState: client.cache.extract(),
    },
    revalidate: 1,
  }
}
```

then in your `<ApolloProvider />` inside of `_app.js` you pass the results of that query to it

```jsx
export default function App({ Component, pageProps }) {
  const apolloClient = useApollo(pageProps.initialApolloState)

  return (
    <ApolloProvider client={apolloClient}>
      <Component {...pageProps} />
    </ApolloProvider>
  )
}
```
